File: AnimalsManagementUI/src/app/Animal/animal.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Animal } from './animal.model';

@Injectable({
  providedIn: 'root'
})
export class AnimalService {
  private apiUrl = 'https://localhost:7004/api/animals';

  constructor(private http: HttpClient) { }

  getAnimals(): Observable<Animal[]> {
    return this.http.get<Animal[]>(this.apiUrl);
  }

  addAnimal(animal: Animal): Observable<Animal> {
    return this.http.post<Animal>(this.apiUrl, animal);
  }

  removeAnimal(name: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${name}`);
  }
}

File: AnimalManagementAPI/Services/AnimalService.cs
using AnimalManagementAPI.Model;

namespace AnimalManagementAPI.Services
{
    public class AnimalService
    {
        private readonly List<Animal> _animals;

        public AnimalService()
        {
            _animals = new List<Animal>();
        }

        public List<Animal> GetAllAnimals()
        {
            return _animals;
        }

        public Animal GetAnimalById(string name)
        {
            return _animals.FirstOrDefault(a => a.Name == name);
        }

        public void AddAnimal(Animal animal)
        {
            _animals.Add(animal);
        }

        public bool DeleteAnimal(string name)
        {
            var animalToRemove = _animals.FirstOrDefault(a => a.Name == name);
            if (animalToRemove != null)
            {
                _animals.Remove(animalToRemove);
                return true;
            }
            return false;
        }
    }
}
